python/ray/dag/tests/test_py_obj_scanner.py

```diff
@@ -31,3 +32,35 @@ def test_not_serializing_objects():
 
     replaced = scanner.replace_nodes({obj: 1 for obj in found})
     assert replaced == [not_serializable, {"key": 1}]
+
+
+def test_scanner_gc():
+    """Test gc collect after the delete instances[id] called"""
```

Comment:
Link to this PR in the docstring please.

Reply:
Did you verify that this test failed prior to the fix?

Reply:
Prior to the fix, we don't remove the scanner in the _instance at all so the test will fail.